apps/crn-server/test/utils/analytics/collaboration.test.ts

```diff
@@ -0,0 +1,219 @@
+    ).toBe(false);
+  });
+
+  it('returns true if one team is the same and no lab is the same', () => {
+    const testAuthor = {
+      id: 'id',
+      teams: [referenceTeam, { sys: { id: 'team-b' } }],
+      labs: [
+        {
+          sys: { id: 'lab-3' },
+        },
+      ],
+    };
+    expect(
+      checkSameTeamDifferentLab(referenceTeam, referenceLabs, testAuthor),
+    ).toBe(true);
+  });
+});
+
+describe('findMatchingAuthors', () => {
+  const referenceId = 'reference-id';
+  const referenceLabs: EntityWithId[] = [];
+  const referenceTeam = { sys: { id: 'reference-team' } };
+  const referenceLab = { sys: { id: 'reference-lab' } };
+  const authorList: Author[] = [];
+
+  const defaultData = {
+    referenceId,
+    referenceLabs,
+    referenceTeam,
+    authorList,
+  };
+
+  it('returns correct flags when there is no authors', () => {
+    expect(findMatchingAuthors(defaultData)).toEqual({
+      differentTeamFlag: false,
+      sameTeamDifferentLabFlag: false,
+    });
+  });
+
+  it('returns correct flags when no matching author is found', () => {
+    const data = {
+      ...defaultData,
+      referenceLabs: [referenceLab],
+      authorList: [
+        {
+          id: 'id',
+          teams: [referenceTeam, { sys: { id: 'team-b' } }],
+          labs: [referenceLab],
+        },
+      ],
+    };
+
+    expect(findMatchingAuthors(data)).toEqual({
+      differentTeamFlag: false,
+      sameTeamDifferentLabFlag: false,
+    });
+  });
+
+  it('returns differentTeamFlag true when appropriate author is found', () => {
+    const data = {
+      ...defaultData,
+      referenceLabs: [referenceLab],
+      authorList: [
+        {
+          id: 'id',
+          teams: [{ sys: { id: 'team-b' } }],
+          labs: [referenceLab],
+        },
+      ],
+    };
+
+    expect(findMatchingAuthors(data)).toEqual({
+      differentTeamFlag: true,
+      sameTeamDifferentLabFlag: false,
+    });
+  });
+
+  it('returns sameTeamDifferentLabFlag true when appropriate author is found', () => {
+    const data = {
+      ...defaultData,
+      referenceLabs: [referenceLab],
+      authorList: [
+        {
+          id: 'id',
+          teams: [referenceTeam, { sys: { id: 'team-b' } }],
+          labs: [{ sys: { id: 'lab-b' } }],
+        },
+      ],
+    };
+
+    expect(findMatchingAuthors(data)).toEqual({
+      differentTeamFlag: false,
+      sameTeamDifferentLabFlag: true,
+    });
+  });
+});
+
+describe('getCollaborationCounts', () => {
+  const flags = (
+    differentTeamFlag: boolean,
+    sameTeamDifferentLabFlag: boolean,
+  ) => ({
+    differentTeamFlag,
+    sameTeamDifferentLabFlag,
+  });
+
+  it('sums each collaboration count properly', () => {
+    const data = [flags(false, false), flags(true, false), flags(false, true)];
+    expect(getCollaborationCounts(data)).toEqual({
+      acrossTeamCount: 1,
+      withinTeamCount: 1,
+    });
+  });
+});
+
+describe('getUserCoproductionItems ', () => {
```

Comment:
I think it would be nice to add the test cases from the JIRA tickets if possible. Here or in apps/crn-server/test/data-providers/contentful/analytics/coproduction.analytics.data-provider.test.ts :)